Source/Fuse.PushNotifications/Android/Impl.uno

```diff
@@ -299,6 +308,25 @@ namespace Fuse.PushNotifications
 				.setAutoCancel(true)
 				.setContentIntent(pendingIntent);
 
+			if(notificationStyle != null && !notificationStyle.isEmpty()){
+
+				if(notificationStyle == BIGTEXTSTYLE || notificationStyle.equals(BIGTEXTSTYLE)){
+
+					notificationBuilder.setStyle(new NotificationCompat.BigTextStyle()
+						.bigText(bigBody)
+						.setBigContentTitle(bigTitle));
+
+				}else if(notificationStyle == BIGPICTURESTYLE || notificationStyle.equals(BIGPICTURESTYLE)){
+
+					notificationBuilder.setStyle(new NotificationCompat.BigPictureStyle()
+						.bigPicture(new ImageDownloaderTask().doInBackground(featured_image))
```

Comment:
I'm a bit confused by this line. You make a ASync task (which is ace as we don't want to block) but then you call the `doInBackground` method directly rather than using `execute`.
I think that, when using a http image you will want to delay the creation of the notification until `onPostExecute`. Also as images are often local it would make sense to support those.

Maybe use uno api to download http/https paths to the apps internal storage and then this java code only has to handle local paths (and then it's fine to just block). This is also nice as we don't need to duplicate download functionality we already have.

Thoughts?

Reply:
I'm sorry, I'm not good using Uno then I've got some doubts.
What do you mean about `execute`? Is there a method for that?
How can I delay the creation of the notification?

I'm gonna investigate how to uso more Uno than Java for the download of the images and the local support. Where should the images be saved? Is there a specific folder for that as cache in build folder?

Reply:
Hi Clebal, no worries about not knowing so much uno. You're doing grand. The issue in this case is in the Java code. `ASyncTask` is for running things on a separate thread. You instantiate it and then call `execute` on it which then starts the new thread and runs the code, you can find more details [here](https://developer.android.com/reference/android/os/AsyncTask.html). 

I was thinking that maybe its best to just assume the path is always to a local file and then user can just use `fetch` to `XMLHttpRequest` to get the file. 
Another reason for this is that every time we send a notification it would download the file again as there is no obvious place to cache the image data here.

Reply:
But if the user has to download the image using `fetch` and in `featured_image` parameter he send the path, what about if the user doesn't know the URL? 

For example, I'm using this to make a notification with BigPictureStyle that use an image from URL, and everytime a blog publish an article, I receive a notification. I don't want to cached those image, because it's just for one use.

I think that it would be better if I add the option to use the path. But URL option should me mantained (using Uno, for sure).

Reply:
Hmm if the address is coming from the notifications then in that case we will have to use the java downloading code because when the notification is recieved the app may not be running so Uno wont be started yet.

So I guess it's back to the AsyncTask issue.